### ISSUE TITLE
Update being skipped on ColorOS

### ISSUE THREAD

**Issue by build3r** [microsoft/appcenter-sdk-android#1547]
### InApp updates not working

The Inapp updates do not work if the user installed the app using ColorOS's file manager (com.coloros.filemanager)

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. Install the app from a file manager **not** listed in the [LOCAL_STORES](https://github.com/microsoft/appcenter-sdk-android/blob/9e7bae9cfdfa69aa4eac5d855c6776f64422dd3f/sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/InstallerUtils.java#L44)
2. Distribute an app update
3. The SDK skips the update.

### Suggestion
The LOCAL_STORES is not exhaustive and is difficult to include all installers. On the other hand the Android Markets are far fewer. We can maintain a negative list to detect if the app was downloaded from an App store.
The other way is to add an extra heuristic approach like `packageInstaller.contains("filemanager")`

### **Details**

1. Which SDK version are you using?
    - e.g. 4.2.0
2. Which OS version did you experience the issue on?
    - e.g. Android 9
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - Oppo, Vivo Device with ColorOS
4. What third party libraries are you using?
    - Not Related
5. Logs
```
Resetting workflow on entering foreground.
D/AppCenterDistribute: Resume distribute workflow...
D/AppCenterDistribute: InstallerPackageName=com.coloros.filemanager
I/AppCenterDistribute: Not checking in app updates as installed from a store.
D/AppCenter: triggerIngestion(group_core) pendingLogCount=1
D/AppCenter: Trying to get 1 logs from the Persistence database for group_core
D/AppCenter: Returning 1 log(s) with an ID, 9c97b9fe-1523-4406-8764-edc4367e33d1
D/AppCenter: The SID/ID pairs for returning log(s) is/are:
D/AppCenter: 	null / 2
D/AppCenter: ingestLogs(group_core,9c97b9fe-1523-4406-8764-edc4367e33d1) pendingLogCount=0
D/AppCenter: checkPendingLogs(group_core) pendingLogCount=0 batchTimeInterval=3000
V/AppCenter: Calling https://in.appcenter.ms/logs?api-version=1.0.0...
V/AppCenter: Headers: {Install-ID=dca71fb7-989f-4fb1-9efe-f55541f3e72a, App-Secret=****************************daaaab07, Content-Type=application/json}
V/AppCenter: {
V/AppCenter: HTTP response status=200 payload={"status":"Success","validDiagnosticsIds":[],"throttledDiagnosticsIds":[],"correlationId":"585ea5a6-57bd-45d5-9c8f-144fdbbd8c80"}
D/AppCenter: Deleting logs from the Persistence database for group_core with 9c97b9fe-1523-4406-8764-edc4367e33d1
D/AppCenter: The IDs for deleting log(s) is/are:
D/AppCenter: 	2
D/AppCenter: checkPendingLogs(group_core) pendingLogCount=0 batchTimeInterval=3000
D/AppCenterDistribute: Resetting workflow on entering foreground.
D/AppCenterDistribute: Resume distribute workflow...
I/AppCenterDistribute: Not checking in app updates as installed from a store.
D/AppCenterDistribute: Resetting workflow on entering foreground.
D/AppCenterDistribute: Resume distribute workflow...
I/AppCenterDistribute: Not checking in app updates as installed from a store.
```

**Comment by DmitriyKirakosyan** [comment]
Hi @build3r 

Thank you for reaching out to us!

Could you add more details on how the installation process looks like for ColorOS?
Do you install the app from the App Center portal?
Is it a default installer that OS uses to install your app from the App Center portal?

**Comment by build3r** [comment]
Our users share the apk with their friends via shareit or xender or whatsapp. This apk is saved on the device.
The user can access the apk file from the default file manager which is com.coloros.filemanager on ColorOS used by Oppo and Vivo OEMs.
When the app is installed by the filemanager the installer pakage is file manager instead of Android package installer. 

This is not a issue if the app is installed from chrome by downloading apk from website.

As you can see from the logs the update fails because the `com.coloros.filemanager` package is not in the list LOCAL_STORES hash map.
Quick fix would be add it to the list.
If possible I can query my products analytics to get any other package installers which are being used.

**Comment by AnastasiaKubova** [comment]
Hi @build3r ! This looks like expected behavior. If you install your app not via the install portal then in-app updates will not work. Please see [guideline](https://docs.microsoft.com/en-us/appcenter/sdk/distribute/android#how-do-in-app-updates-work) how to work in-app updates.

**Comment by build3r** [comment]
@AnastasiaKubova
Our app is currently on public track and we distribute it outside of Playstore, via app Center and our own website. And app center is used push updates.
Looking at sdk behaviour the assumption is the app on public tracks will be updated from app center if they are not installed from Playstore(due to its policy on outside updates).
Am I wrong in this assumption?

**Comment by AnastasiaKubova** [comment]
Yes, you are right! If you use a public group then in-app updates should work if you distribute your app outside of Play Store. App Center SDK currently supports the [next](https://github.com/microsoft/appcenter-sdk-android/blob/9e7bae9cfdfa69aa4eac5d855c6776f64422dd3f/sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/InstallerUtils.java#L44) local storages. I label this issue as a feature request. But feel free to create a PR if you have a solution for it.

**Comment by AnastasiaKubova** [comment]
Hi there! The fix has been released in the [4.3.1 version](https://github.com/microsoft/appcenter-sdk-android/releases/tag/4.3.1) of App Center SDK.
